File: api/ResourceManagerPro/src/main/java/com/cuea/rmp/shared/application/ApiResponse.java
package com.cuea.rmp.shared.application;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;

/**
 * Uniform response envelope returned by every endpoint.
 *
 * @param success    whether the request succeeded
 * @param message    human-readable message (nullable on plain success)
 * @param data       payload on success (nullable on failure)
 * @param errorCode  stable machine code on failure (nullable on success)
 * @param timestamp  server time the response was produced (UTC)
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiResponse<T>(
        boolean success,
        String message,
        T data,
        String errorCode,
        Instant timestamp
) {

    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(true, null, data, null, Instant.now());
    }

    public static <T> ApiResponse<T> ok(T data, String message) {
        return new ApiResponse<>(true, message, data, null, Instant.now());
    }

    public static <T> ApiResponse<T> fail(String message) {
        return new ApiResponse<>(false, message, null, null, Instant.now());
    }

    public static <T> ApiResponse<T> fail(String message, String errorCode) {
        return new ApiResponse<>(false, message, null, errorCode, Instant.now());
    }
}

File: ui/app/src/main/java/com/cuea/rmp/mobile/core/network/ApiResponse.kt
package com.cuea.rmp.mobile.core.network

import kotlinx.serialization.Serializable

@Serializable
data class ApiResponse<T>(
    val success: Boolean,
    val message: String? = null,
    val data: T? = null,
    val errorCode: String? = null,
    val timestamp: String? = null
)

@Serializable
data class PageResult<T>(
    val content: List<T>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int
)

